src/components/related-grid/page-reducer.ts

```diff
@@ -0,0 +1,13 @@
+import {PageAction} from './page-action';
+
+const PageReducer = (state: {prevPage: number; currPage: number; nextPage: number}, action: PageAction) => {
+  const {prevPage, currPage, nextPage} = state;
+  const offset = action === PageAction.NEXT ? 1 : -1;
```

Comment:
don't you get here for PageAction.NOTHING? if so, offset should be 0 no?

Reply:
NOTHING is just used as the default state, i don't call the reducer in that case